0.download-data/tcga_joint_representation_data.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""tcga-joint-representation-data.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1lJo-gGg3wPVnpMygd8UG3tKoA1y6OVV8
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import os
+import seaborn as sns
+pd.set_option('display.max_colwidth', None)
+
+#Direct URL
+#tcga_url = 'https://api.gdc.cancer.gov/cases?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22TCGA%22%5D%7D%7D%5D%7D'
+
+import requests
+import json
+
+fields = [
+    "case_id",
+    "submitter_id",
+    "project.program.name",
+    "project.project_id",
+    "demographic.ethnicity",
+    "demographic.gender",
+    "demographic.race",
+    "primary_diagnosis",
+    "diagnoses.icd_10_code",
+    "diagnoses.age_at_diagnosis",
+    "disease_type",
+    "primary_site",
+    "demographic.year_of_birth",
+    "demographic.year_of_death",
+    "demographic.days_to_death",
+    "demographic.vital_status"
+    ]
+
+fields = ",".join(fields)
+
+cases_endpt = "https://api.gdc.cancer.gov/cases"
+
+filters = {
+            "op": "in",
+            "content":{
+            "field": "project.program.name",
+            "value": ["TCGA"]
+            }
+    }
+
+# With a GET request, the filters parameter needs to be converted
+# from a dictionary to JSON-formatted string
+
+params = {
+    "filters": json.dumps(filters),
+    "fields": fields,
+    "format": "JSON",
+    "size": "11500"
+    }
+
+response = requests.post(cases_endpt, headers = {"Content-Type": "application/json"}, json = params)
+#print(response.content.decode("utf-8"))
+
+data = json.loads(response.content.decode('utf-8'))
+
+df = pd.json_normalize(data['data'], 'hits')
```

Comment:
Please save this data frame and add it to this repository. Probably in a file named: `0.download-data/data/metadata.tsv`

It will be much easier to use as a file saved in this repo than having to load the json every time!

See [`pandas.DataFrame.to_csv`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html).